internal/service/advancedcluster/resource_advanced_cluster_test.go

```diff
@@ -866,6 +867,128 @@ func TestAccClusterAdvancedCluster_priorityNewSchema(t *testing.T) {
 	})
 }
 
+func TestAccAdvancedCluster_oldToNewSchemaWithAutoscalingDisabled(t *testing.T) {
+	acc.SkipIfTPFAdvancedCluster(t)
+	var (
+		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName = acc.RandomProjectName()
+		clusterName = acc.RandomClusterName()
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 acc.PreCheckBasicSleep(t, nil, orgID, projectName),
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             acc.CheckDestroyCluster,
+		Steps: []resource.TestStep{
+			{
+				Config: configShardedTransitionOldToNewSchema(orgID, projectName, clusterName, false, false),
+				Check:  checkIndependentShardScalingMode(clusterName, "CLUSTER"),
+			},
+			{
+				Config: configShardedTransitionOldToNewSchema(orgID, projectName, clusterName, true, false),
+				Check:  checkIndependentShardScalingMode(clusterName, "CLUSTER"),
+			},
+		},
+	})
+}
+
+func TestAccAdvancedCluster_oldToNewSchemaWithAutoscalingEnabled(t *testing.T) {
+	acc.SkipIfTPFAdvancedCluster(t)
+	var (
+		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName = acc.RandomProjectName()
+		clusterName = acc.RandomClusterName()
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
```

Comment:
Effectively, all these TestCases are the same, except for the config arguments and CLUSTER/SHARD. I would love to see a refactoring re-using the test case to avoid all the boilerplate